daft/dataframe/dataframe.py

```diff
@@ -1312,7 +1304,18 @@ def _wildcard_inputs_to_expressions(self, columns: tuple[ManyColumnsInputType, .
         column_input: Iterable[ColumnInputType] = columns[0] if len(columns) == 1 else columns  # type: ignore
         return column_inputs_to_expressions(column_input)
 
-    def __getitem__(self, item: Union[slice, int, str, Iterable[Union[str, int]]]) -> Union[Expression, "DataFrame"]:
+    if TYPE_CHECKING:
+
+        @overload
+        def __getitem__(self, item: int) -> Expression: ...
+        @overload
+        def __getitem__(self, item: str) -> Expression: ...
+        @overload
+        def __getitem__(self, item: slice) -> "DataFrame": ...
+        @overload
+        def __getitem__(self, item: Iterable) -> "DataFrame": ...  # type: ignore
```

Comment:
Had to do this because strings in Python are `Iterable[str]`, so mypy kept complaining about conflicting type definitions. See https://github.com/python/mypy/issues/11001